src/cleanfigure.m

```diff
@@ -924,19 +1050,46 @@ function removeNaNs(meta, handle)
   data(id_remove,:) = [];
 
   % Set the new data.
-  set(handle, 'XData', data(:,1));
-  set(handle, 'YData', data(:,2));
-  if isAxis3D(meta.gca)
-  	set(handle, 'zData', data(:,3));
+  set(handle, 'XData', data(:, 1));
+  set(handle, 'YData', data(:, 2));
+  if is3D
+	set(handle, 'ZData', data(:, 3));
   end
 end
 % ==========================================================================
-function [bottomLeft, topLeft, bottomRight, topRight] = corners(xLim, yLim)
-    % Determine the corners of the axes as defined by xLim and yLim
-    bottomLeft  = [xLim(1); yLim(1)];
-    topLeft     = [xLim(1); yLim(2)];
-    bottomRight = [xLim(2); yLim(1)];
-    topRight    = [xLim(2); yLim(2)];
+function [bottomLeft, topLeft, bottomRight, topRight] = corners2D(xLim, yLim)
+  % Determine the corners of the axes as defined by xLim and yLim
+  bottomLeft  = [xLim(1), yLim(1)];
+  topLeft     = [xLim(1), yLim(2)];
+  bottomRight = [xLim(2), yLim(1)];
+  topRight    = [xLim(2), yLim(2)];
+end
+% ==========================================================================
+function corners = corners3D(xLim, yLim, zLim)
+    % Determine the corners of the 3D axes as defined by xLim, yLim, and
+    % zLim
+
+    % Lower square of the cube
+    lowerBottomLeft  = [xLim(1), yLim(1), zLim(1)];
```

Comment:
Also in a different note, it would considerably change the syntax corners2D is used. The only way around would be to have 8 outputs in corners3D which is kind of painfull.